bundle/config/validate/single_node_cluster.go

```diff
@@ -0,0 +1,174 @@
+		Detail:    singleNodeWarningDetail,
+		Locations: l,
+		Paths:     []dyn.Path{p},
+	}
+	profile, ok := spec.SparkConf["spark.databricks.cluster.profile"]
+	if !ok {
+		return invalidSingleNodeWarning
+	}
+	master, ok := spec.SparkConf["spark.master"]
+	if !ok {
+		return invalidSingleNodeWarning
+	}
+	resourceClass, ok := spec.CustomTags["ResourceClass"]
+	if !ok {
+		return invalidSingleNodeWarning
+	}
+
+	if profile == "singleNode" && strings.HasPrefix(master, "local") && resourceClass == "SingleNode" {
+		return nil
+	}
+
+	return invalidSingleNodeWarning
+}
+
+func validateSingleNodePipelineCluster(spec pipelines.PipelineCluster, l []dyn.Location, p dyn.Path) *diag.Diagnostic {
+	if spec.NumWorkers > 0 || spec.Autoscale != nil {
+		return nil
+	}
+
+	if spec.PolicyId != "" {
+		return nil
+	}
+
+	invalidSingleNodeWarning := &diag.Diagnostic{
+		Severity:  diag.Warning,
+		Summary:   singleNodeWarningSummary,
+		Detail:    singleNodeWarningDetail,
+		Locations: l,
+		Paths:     []dyn.Path{p},
+	}
+	profile, ok := spec.SparkConf["spark.databricks.cluster.profile"]
+	if !ok {
+		return invalidSingleNodeWarning
+	}
+	master, ok := spec.SparkConf["spark.master"]
+	if !ok {
+		return invalidSingleNodeWarning
+	}
+	resourceClass, ok := spec.CustomTags["ResourceClass"]
+	if !ok {
+		return invalidSingleNodeWarning
+	}
+
+	if profile == "singleNode" && strings.HasPrefix(master, "local") && resourceClass == "SingleNode" {
+		return nil
+	}
+
+	return invalidSingleNodeWarning
+}
+
+func (m *singleNodeCluster) Apply(ctx context.Context, rb bundle.ReadOnlyBundle) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+
+	// Interactive clusters
+	for k, r := range rb.Config().Resources.Clusters {
+		p := dyn.NewPath(dyn.Key("resources"), dyn.Key("clusters"), dyn.Key(k))
+		l := rb.Config().GetLocations("resources.clusters." + k)
+
+		d := validateSingleNodeCluster(r.ClusterSpec, l, p)
+		if d != nil {
+			diags = append(diags, *d)
+		}
+	}
+
+	// Job clusters
+	for jobK, jobV := range rb.Config().Resources.Jobs {
+		for i, clusterV := range jobV.JobSettings.JobClusters {
+			p := dyn.NewPath(dyn.Key("resources"), dyn.Key("jobs"), dyn.Key(jobK), dyn.Key("job_clusters"), dyn.Index(i))
+			l := rb.Config().GetLocations(fmt.Sprintf("resources.jobs.%s.job_clusters[%d]", jobK, i))
+
+			d := validateSingleNodeCluster(&clusterV.NewCluster, l, p)
+			if d != nil {
+				diags = append(diags, *d)
+			}
+		}
+	}
+
+	// Job task clusters
+	for jobK, jobV := range rb.Config().Resources.Jobs {
+		for i, taskV := range jobV.JobSettings.Tasks {
+			if taskV.NewCluster == nil {
+				continue
+			}
+
+			p := dyn.NewPath(dyn.Key("resources"), dyn.Key("jobs"), dyn.Key(jobK), dyn.Key("tasks"), dyn.Index(i), dyn.Key("new_cluster"))
+			l := rb.Config().GetLocations(fmt.Sprintf("resources.jobs.%s.tasks[%d].new_cluster", jobK, i))
+
+			d := validateSingleNodeCluster(taskV.NewCluster, l, p)
+			if d != nil {
+				diags = append(diags, *d)
+			}
+		}
+	}
+
+	// Pipeline clusters
```

Comment:
Yeah, I confirmed with the UC team that single-node clusters can be used with DLT. The rules that apply should be the same since the pipeline cluster interface mirrors the cluster API. Thus it's a safe bet that they rely on the clusters API.